core/gui/src/app/workspace/service/workflow-result-export/workflow-result-export.service.ts

```diff
@@ -48,12 +54,86 @@ export class WorkflowResultExportService {
+          .filter(
+            operatorId =>
+              this.workflowResultService.hasAnyResult(operatorId) ||
+              this.workflowResultService.getResultService(operatorId)?.getCurrentResultSnapshot() !== undefined
+          ).length > 0;
+
+      // check if there are any results to export on all operators (either paginated or snapshot)
+      this.hasResultToExportOnAllOperators =
+        isNotInExecution(this.executeWorkflowService.getExecutionState().state) &&
+        this.workflowActionService
+          .getTexeraGraph()
+          .getAllOperators()
+          .map(operator => operator.operatorID)
+          .filter(
+            operatorId =>
+              this.workflowResultService.hasAnyResult(operatorId) ||
+              this.workflowResultService.getResultService(operatorId)?.getCurrentResultSnapshot() !== undefined
+          ).length > 0;
+    });
+  }
+
+  /**
+   * Export the operator results as files.
+   * If multiple operatorIds are provided, results are zipped into a single file.
+   */
+  exportOperatorsResultAsFile(download_all: boolean = false): void {
+    let operatorIds: string[];
+    if (!download_all)
+      operatorIds = [...this.workflowActionService.getJointGraphWrapper().getCurrentHighlightedOperatorIDs()];
+    else
+      operatorIds = this.workflowActionService
+        .getTexeraGraph()
+        .getAllOperators()
+        .map(operator => operator.operatorID);
+
+    const resultObservables: Observable<{ filename: string; blob: Blob }[]>[] = [];
+
+    operatorIds.forEach(operatorId => {
+      const resultService = this.workflowResultService.getResultService(operatorId);
+      const paginatedResultService = this.workflowResultService.getPaginatedResultService(operatorId);
+
+      if (paginatedResultService) {
+        const observable = this.fetchAllPaginatedResultsAsCSV(paginatedResultService, operatorId);
+        resultObservables.push(observable);
+      } else if (resultService) {
+        const observable = this.fetchVisualizationResultsAsHTML(resultService, operatorId);
+        resultObservables.push(observable);
+      } else {
+        // No result service available for this operatorId
+        this.notificationService.warning(`No results available for operator ID: ${operatorId}`);
+      }
+    });
+
+    if (resultObservables.length === 0) {
+      return;
+    }
+
+    forkJoin(resultObservables).subscribe(filesArray => {
+      const files = filesArray.flat();
+
+      if (files.length === 1) {
+        // Only one file, save it directly
+        this.fileSaverService.saveAs(files[0].blob, files[0].filename);
+      } else if (files.length > 1) {
+        // Multiple files, zip them
+        const zip = new JSZip();
+        files.forEach(file => {
+          zip.file(file.filename, file.blob);
+        });
+        zip.generateAsync({ type: "blob" }).then((zipBlob: string | Blob) => {
+          const zipFilename = `results_${new Date().getTime()}.zip`;
```

Comment:
I think it is better to name the zip file using workflow id and workflow name. (e.g., result_[workflow id]_[workflow name].zip).

Reply:
Done